pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/kit/KitFinalSentBaseUseCase.java

```diff
@@ -49,13 +49,18 @@ protected void triggerEvents(String ddpLabel, KitRequestShipping kitRequestShipp
         }
         if (kitPayload.getDdpInstanceDto().isESUpdatePossible()) {
             try {
-                UpsertPainlessFacade.of(DBConstants.DDP_KIT_REQUEST_ALIAS, kitRequestShipping,
-                        kitPayload.getDdpInstanceDto(), "ddpLabel", "ddpLabel",
-                        ddpLabel, new PutToNestedScriptBuilder()).export();
+                //to avoid updating old deactivated record in ES because of search by label .. update only if NO de-activated record exists with same label
+                //check if de-activated kit request exists for same kit-request-id and skip ES update
```

Comment:
ES is updated anyway with new kitRequestShipping recod in the immediate KitFinalScanUseCase.writeSampleSentToES call during the final-scan

Reply:
Not sure I follow what is going on here.
Won't anything not written to ES here just get written on the next DSM export?

Reply:
As long as the data in the DB doesn't change, ES exported data shouldn't change. 
Here we're avoiding updating existing kitRequestShipping element/record in this sample participant ES index
https://f7df559584f84f2cab27de822618fb89.us-central1.gcp.cloud.es.io:9243/participants_structured.cmi.cmi-pancan/_search?pretty&q=profile.hruid:PB632F

Avoid updating below existing kitRequestShipping record 
{
                "deactivationReason" : "testDeactivate-1718883724841",
                "ddpLabel" : "N9PPHR8QLIWE4I5",
                "bspCollaboratorSampleId" : "Project_PB632F_SALIVA",
                "dsmKitId" : 59239,
                "deactivatedDate" : 1718883725126,
                "express" : false,
                "error" : false,
                "message" : null,
                "bspCollaboratorParticipantId" : "Project_PB632F",
                "createdBy" : "SYSTEM",
                "ddpInstanceId" : 24,
                "noReturn" : false,
                "kitTypeName" : "SALIVA",
                "ceOrder" : false,
                "dsmKitRequestId" : 59604
              },
              
             with new element/kit created for same label: N9PPHR8QLIWE4I5
             {
                "kitLabel" : "SS_PB632F_SS01",
                "ddpLabel" : "N9PPHR8QLIWE4I5",
                "bspCollaboratorSampleId" : "Project_PB632F_SALIVA",
                "dsmKitId" : 60554,
                "express" : false,
                "error" : true,
                "message" : "To: Connection to the carrier timed out. Please try again later.",
                "bspCollaboratorParticipantId" : "Project_PB632F",
                "scanDate" : 1719889806249,
                "createdBy" : "SYSTEM",
                "trackingReturnId" : "Track_PB632F_SS01",
                "ddpInstanceId" : 24,
                "noReturn" : false,
                "kitTypeName" : "SALIVA",
                "ceOrder" : false,
                "labelDate" : 1719889558467,
                "dsmKitRequestId" : 59604
              },
             
             
             Participant data without the fix
https://f7df559584f84f2cab27de822618fb89.us-central1.gcp.cloud.es.io:9243/participants_structured.cmi.cmi-pancan/_search?pretty&q=profile.hruid:PAVZC3
Search for label: X8ANWLNGOHMZRMO
You can see that both kitRequestShipping elements have same data (new kit info)